src/rgw/rgw_sal_motr.cc

```diff
@@ -4012,6 +4337,46 @@ int MotrStore::init_metadata_cache(const DoutPrefixProvider *dpp,
   return 0;
 }
 
+void MotrStore::addb_map_init(void)
+{
+  for (uint64_t idx = 0; idx < g_rgw_to_addb_idx_func_name_map_size; idx++) {
+    addb_func_to_idx_map[g_rgw_to_addb_idx_func_name_map[idx]] = idx;
+#if defined ADDB_DEBUG
+    ldout(cctx, 0) << "[MM] Registering func: <" << g_rgw_to_addb_idx_func_name_map[idx] << "> with ID: " << idx << dendl;
+#endif
+  }
+}
+
+uint64_t MotrStore::addb_func_to_idx(const std::string& funcname)
```

Comment:
How does it work, and why is this needed?

Reply:
As ADDB only allows to "log" parameters of uint64_t type, but not strings, lookup table is implemented here.
This function takes function name parameter (__PRETTY_FUNCTION__ macro), finds corresponding index in the  `const char* g_rgw_to_addb_idx_func_name_map[] ` map and returns an index of that function name that will be logged.
ADDB sample added to ADDB stob contains probe_type (i.e. type of request), request id and function idx.
During ADDB stob processing time, m0addb2dump finds probe information from ADDB plugin (rgw_addb_plugin.so) and uses the same map to match function idx with the function name. Function name in its order is added in human-readable way in resulting text dump file.

Reply:
In motr we just log the address of the function (m0_ptr_wrap(&func)) and then in addb2/dump.c use sym() and "-l path-to-shared-library".

Reply:
So why cannot we log the address of the function here also, similar like we do in motr?

Reply:
Two reasons:
1. I had no clue about m0_ptr_wrap() and function name resolving. Thanks @nikitadanilov for pointing the right direction. S3server code implements the same function <-> idx map as implemented here, and just reused the code.
2. @andriytk Does RGW build system generate shared (or static) library? If yes, what is it? Can libbfd() deal with C++ symbols? I suggest to take function address-name resolution as another ticket (and PR)